docs/tutorial/01-getting-started.md

```diff
@@ -0,0 +1,147 @@
+- `Administratable` provides functions like `require_admin()` to manage who can preform administrative actions
+- `Upgradable` allows the contract to be upgraded to new versions while preserving its state
+
+We're also importing essential items from Stellar's Soroban SDK, and we'll explain each as we get to them. You'll see that many of them replace items from the standard library but are designed for use in Soroban's environment. And the first is `contract`:
+
+```rust
+#[contract]
+pub struct GuessTheNumber;
+
+#[contractimpl]
+impl Administratable for GuessTheNumber {}
+
+#[contractimpl]
+impl Upgradable for GuessTheNumber {}
+```
+
+The `#[...]` syntax in Rust is called an [attribute](https://doc.rust-lang.org/reference/attributes.html). It's a way to label code for the compiler to handle it with special instructions. *Inner* attributes (with the `#!`) apply to the scope they're within, and *Outer* attributes (just the `#`) apply to the next line.
+
+Here defining a [struct](https://doc.rust-lang.org/book/ch05-01-defining-structs.html) (a "structure" to hold values) and applying attributes of a Stellar smart contract. Then we'll use the pre-defined [traits](https://doc.rust-lang.org/book/ch10-02-traits.html) we just explained to add more functionality to our contract without actually having to write it ourselves.
+
+```rust
+const THE_NUMBER: Symbol = symbol_short!("n");
+```
+
+Now the most important part of our contract: the number! This line creates a key for storing and retrieving contract data. A `Symbol` is a short string type (max 32 characters) that is more optimized for use on the blockchain. And we're using the `symbol_short` macro for an even smaller key (max 9 characters). As a contract author, you want to use tricks like this to lower costs as much as you can.
+
+```rust
+#[contractimpl]
+impl GuessTheNumber {
+```
+
+Let's `impl`ement our contract's functionality.
+
+```rust
+    pub fn __constructor(env: &Env, admin: &Address) {
+        Self::set_admin(env, admin);
+    }
+```
+
+A contract's `constructor` runs when it is deployed. In this case, we're saying who has access to the admin functions. We don't want anyone to be able to reset our number, do we?!
+
+```rust
+    /// Update the number. Only callable by admin.
+    pub fn reset(env: &Env) {
+        Self::require_admin(env);
+        let new_number: u64 = env.prng().gen_range(1..=10);
+        env.storage().instance().set(&THE_NUMBER, &new_number);
+    }
+```
+
+And here is the reset function. Note that we use `require_admin()` here so only you can run this function. It generates a random number between 1 and 10 and uses our key to store it.
+
+```rust
+    /// Guess a number between 1 and 10
+    pub fn guess(env: &Env, a_number: u64) -> bool {
+        a_number == env.storage().instance().get::<_, u64>(&THE_NUMBER).unwrap()
+    }
+}
```

Comment:
Because of the contracts refactor, you can reference public/private functions if you like. Not sure if it's necessary, but it's an option